### PR TITLE
Improve code for getting example inputs for IPEX backend

### PR DESCRIPTION
**Pull request by violetch24**
Signed-off-by: Cheng, Zixuan <zixuan.cheng@intel.com>

## Type of Change

- add torch version limit >=1.12.0
- change input from dict to namedtuple
API not changed

## Description

JIRA ticket: https://jira.devtools.intel.com/browse/NLPTOOLKIU-393

### CODE REVIEW

File: neural_compressor/adaptor/pytorch.py [hunk 1]
```diff
@@ -136,14 +137,16 @@ def get_example_inputs(model, dataloader):  # pragma: no cover
...
             if isinstance(input, dict) or isinstance(input, UserDict):
-                input_tensor = []
+                assert version.release >= Version("1.12.0").release, \
+                "INC support IPEX version >= 1.12.0"
                 if "label" in input.keys():
                     input.pop("label")
-                for key, value in input.items():
+                for key in input.items():
                     if key == "start_positions" or key == "end_positions":
                         continue
```
Comment (PenghuiCheng) [inline neural_compressor/adaptor/pytorch.py:146]: These code seems to not needed

File: neural_compressor/adaptor/pytorch.py [hunk 2]
```diff
@@ -154,14 +157,16 @@ def get_example_inputs(model, dataloader):  # pragma: no cover
...
```
Comment (PenghuiCheng) [inline neural_compressor/adaptor/pytorch.py:166]: remove it?

**Review by PenghuiCheng — APPROVED** [review]

**Comment by PenghuiCheng** [comment]
extension: https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/4281/

**Comment by chensuyue** [comment]
Test pass after remove `datasets` version limit.  
https://inteltf-jenk.sh.intel.com/view/lpot-validation/job/intel-lpot-validation-top-mr-extension/4305/ 
Retest: https://inteltf-jenk.sh.intel.com/view/lpot-validation/job/intel-lpot-validation-top-mr-extension/4311/

**Pull request merged**